include/dlaf/common/index2d.h

```diff
@@ -401,5 +401,11 @@ Size2D<IndexT, Tag> operator+(const Size2D<IndexT, Tag>& lhs, const Size2D<Index
   return Size2D<IndexT, Tag>(lhs.rows() + rhs.rows(), lhs.cols() + rhs.cols());
 }
 
+/// Returns true if @param small is within @param big starting at @param idx
+template <class IndexT, class Tag>
+bool isWithin(Size2D<IndexT, Tag> small, Index2D<IndexT, Tag> idx, Size2D<IndexT, Tag> big) {
```

Comment:
Just a minor suggestion. I like free functions more than member functions, and I'm just wondering if moving also `Index2D::isIn()` function wouldn't be a good idea, just to keep them "close" since they are in some way semantically similar.

Reply:
I can certainly move it out, the issue is that I will have to touch quite a lot of files as the function is used in many places. Perhaps we can leave that to a separate PR.